mlir/lib/Quantum/Transforms/merge_rotation.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define DEBUG_TYPE "merge-rotation"
+
+#include "mlir/Pass/Pass.h"
+#include "llvm/Support/Debug.h"
+
+#include "Catalyst/IR/CatalystDialect.h"
+#include "Quantum/IR/QuantumOps.h"
+
+using namespace llvm;
+using namespace mlir;
+using namespace catalyst::quantum;
+
+namespace catalyst {
+namespace quantum {
+
+#define GEN_PASS_DEF_MERGEROTATIONPASS
+#define GEN_PASS_DECL_MERGEROTATIONPASS
+#include "Quantum/Transforms/Passes.h.inc"
+
+struct MergeRotationPass : impl::MergeRotationPassBase<MergeRotationPass> {
+    using MergeRotationPassBase::MergeRotationPassBase;
+
+    void runOnOperation() final
+    {
```

Comment:
Can you make it such that it does nothing? It is possible that users will be tempted to try it

Reply:
I made it so that we can test the pass options are passed down from frontend to mlir correctly. In added pytest there's a check for the message. Without that there would be no tests for this. I think this is fine as `merge_rotation` will not be added to the frontend documentation in this PR.

Reply:
You can also create a test-pass instead of one that has a specific name that actually means something. :) That way, users wouldn't care about running a test-pass.

Reply:
> You can also create a test-pass instead of one that has a specific name that actually means something

I think we are against having "empty code" in the repo, e.g. second point of [this comment](https://github.com/PennyLaneAI/catalyst/pull/872#issuecomment-2237714359). 

I used merge rotation because it will soon become filled with actual content instead of being a true empty boilerplate. I don't want users to access it already so I did not add a `merge_rotation` frontend API to the documentation.

Reply:
I removed merge_rotation in all public documentation.